lib/memery.rb

```diff
@@ -43,39 +34,61 @@ def memoized?(method_name)
 
     def prepend_memery_module!
       return if defined?(@_memery_module)
-      @_memery_module = Module.new
+      @_memery_module = Module.new do
+        extend MemoizationModule
+      end
       prepend @_memery_module
     end
 
-    def define_memoized_method!(method_name, condition: nil, ttl: nil)
-      visibility = Memery.method_visibility(self, method_name)
-      raise ArgumentError, "Method #{method_name} is not defined on #{self}" unless visibility
+    def define_memoized_method!(*args, **kwargs)
+      @_memery_module.public_send __method__, self, *args, **kwargs
+    end
 
-      method_key = "#{method_name}_#{@_memery_module.object_id}"
+    module MemoizationModule
+      def define_memoized_method!(klass, method_name, condition: nil, ttl: nil)
+        method_key = "#{method_name}_#{object_id}"
 
-      # Change to regular call of `define_method` after Ruby 2.4 drop
-      @_memery_module.send :define_method, method_name, (lambda do |*args, **kwargs, &block|
-        if block || (condition && !instance_exec(&condition))
-          return kwargs.any? ? super(*args, **kwargs, &block) : super(*args, &block)
-        end
+        original_visibility = method_visibility klass, method_name
```

Comment:
Please use brackets in cases when we care about return value.

Reply:
Sorry, what? Do you talk about parentheses? And how they're affecting here?

Reply:
Just a codestyle convention that I like to follow. See https://github.com/airbnb/ruby#returns-val-parens

Reply:
Parentheses can also be called _round brackets_ btw :)

Reply:
> Just a codestyle convention that I like to follow. See https://github.com/airbnb/ruby#returns-val-parens

OK, as you wish. Changed.